File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "brick_imu_v2.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XXYYZZ" // Change XXYYZZ to the UID of your IMU Brick 2.0

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	IMUV2 imu;
	imu_v2_create(&imu, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current quaternion
	int16_t w, x, y, z;
	if(imu_v2_get_quaternion(&imu, &w, &x, &y, &z) < 0) {
		fprintf(stderr, "Could not get quaternion, probably timeout\n");
		return 1;
	}

	printf("Quaternion [W]: %f\n", w/16383.0);
	printf("Quaternion [X]: %f\n", x/16383.0);
	printf("Quaternion [Y]: %f\n", y/16383.0);
	printf("Quaternion [Z]: %f\n", z/16383.0);

	printf("Press key to exit\n");
	getchar();
	imu_v2_destroy(&imu);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"tinkerforge/imu_v2_brick"
	"tinkerforge/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your IMU Brick 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	imu, _ := imu_v2_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current quaternion.
	w, x, y, z, _ := imu.GetQuaternion()

	fmt.Printf("Quaternion [W]: %f\n", float64(w)/16383.0)
	fmt.Printf("Quaternion [X]: %f\n", float64(x)/16383.0)
	fmt.Printf("Quaternion [Y]: %f\n", float64(y)/16383.0)
	fmt.Printf("Quaternion [Z]: %f\n", float64(z)/16383.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}